test/libsolidity/semanticTests/memoryManagement/struct_allocation.sol

```diff
@@ -0,0 +1,24 @@
+contract C {
+    struct S { uint x; uint y; uint z; }
```

Comment:
I'd add a few more test cases:
- Statically-sized arrays: Should work like for structs.
- Dynamically-sized arrays
    - Should not allocate memory at all by default - or do we already have a test for that?
    - The reference itself should be initialized with zero (or a zero pointer?)
- Returning a newly allocated struct or statically-sized array literal from a function.